src/modules/aws_ecr/index.ts

```diff
@@ -626,11 +622,11 @@ class RepositoryPolicyMapper extends MapperBase<RepositoryPolicy> {
     update: (es: RepositoryPolicy[], ctx: Context) => ctx.orm.save(RepositoryPolicy, es),
     delete: (es: RepositoryPolicy[], ctx: Context) => ctx.orm.remove(RepositoryPolicy, es),
     read: async (ctx: Context, id?: string) => {
-      const [repositoryName, region] = id?.split('|') ?? [undefined, undefined];
-      const opts = repositoryName
+      const [repositoryId, region] = id?.split('|') ?? [undefined, undefined];
```

Comment:
This is not correct. When `RepositoryPolicy` `db.read` is called it will be passing the `RepositoryPolicy.generateId()` method as id. The `generateId`, `entityId` and `idFields` methods are handling the cloud identification of the policy based on the `repositoryName`. So here we should keep `repositoryName` as the value retrieved from the `id` and the fix should be in the where clause using `repository: { repositoryName }` instead of `repository: { id: repositoryId }`

Reply:
![right](https://user-images.githubusercontent.com/5357435/208623864-c00e5db8-fc6e-4cff-8087-0b8e2c545bcb.gif)